metastore/src/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java

```diff
@@ -302,23 +302,14 @@ private void open() throws MetaException {
 
     SocketFactory socketFactory = SocketFactory.getDefault();
     Socket socket = null;
+    String hiveHost = null;
     try {
       socket = socketFactory.createSocket();
       socket.setKeepAlive(true);
 
-      String hiveHost = conf.getVar(ConfVars.HIVE_SERVER2_THRIFT_BIND_HOST);
-      if (hiveHost.isEmpty()) {
-        UserGroupInformation loginUser = UserGroupInformation.getLoginUser();
-        if (loginUser != null && loginUser.hasKerberosCredentials()) {
-          hiveHost = SecurityUtil.getHostFromPrincipal(loginUser.getUserName());
-        }
-      }
-
-      if (!hiveHost.isEmpty()) {
-        InetAddress localAddr = NetUtils.getLocalInetAddress(hiveHost);
-        if (localAddr != null) {
-          socket.bind(new InetSocketAddress(localAddr, 0));
-        }
+      UserGroupInformation loginUser = UserGroupInformation.getLoginUser();
+      if (loginUser != null && loginUser.hasKerberosCredentials()) {
+        hiveHost = SecurityUtil.getHostFromPrincipal(loginUser.getUserName());
```

Comment:
What happens in the case that we are not running Kerberos? hiveHost is never set. Is that acceptable?
I'll re-iterate. I am NOT familiar with this code. *Please please please post a patch in the open-source community*. I know its the longer route, but you *have* to take it. Internally forking code has not worked well for most companies.